### PR TITLE
ath10k: update QCA4019 firmware.

### PR DESCRIPTION
**Pull request by vort7u**
With AVM Fritz!Box 4040 and OpenWrt 18.06 RC1 there are many kernel warnings
`kern.warn kernel: [87771.917049] ath10k_ahb a000000.wifi: Invalid VHT mcs 15 peer stats`
and there are disconnections when the connected clients are many, at the moment I tried with 16 clients on 2.4 GHz and 8 on 5 GHZ.

Firmware 3.6 fixes these warnings and the problem of disconnections of some clients.

Signed-off-by: Massimo T. <masnia@tiscali.it>

### CODE REVIEW

**Comment by robimarko** [comment]
Still missing signed of by line in PR and commit

**Comment by blocktrron** [comment]
@mastum Use `git commit --amend --signoff` to create the Signed-Off-By line. Force push to you existing branch with `git push -f`, this will update your PR to match the requirements.

While i also experienced the warnings you mention, i have not experienced any performance or stability improvement by using newer firmware, can you explain which stability issues you were facing?

**Comment by mkresin** [comment]
Commit title should be `ath10k: update QCA4019 firmware`.

> More info on forum  https://forum.lede-project.org/t/solved-ipq40xx-and-ath10k-ahb-a000000-wifi-invalid-vht-mcs-15-peer-stats/16398

Nope, mention in the commit message what issues are fixed with the update. Make it easy for me to merge the PR by providing all relevant informations and don't force me to skim through dozen of forum posts.

**Comment by mastum** [comment]
Modified, thanks to everyone.

**Comment by mkresin** [comment]
> Modified, thanks to everyone.

You have to modify the commit message and not the PR description. Only the commit message will be later seen in the repository.

Please use ```git commit --amend``` to add changes to your commit. Use ```git push --force``` to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

**Comment by mastum** [comment]
Done, thanks.

**Comment by ecsv** [comment]
Warning: you must also merge the newest DFS host confirmation mis-feature support for this firmware:

* https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/net/wireless/ath/ath10k?id=f40105e6747892e8edab94020567c158c9bec0df
* https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/net/wireless/ath/ath10k?id=6f6eb1bcbeff48c875617b800f00f1c5d1b12290

This mis-feature will break the wifi-tx whenever the QCA internal matcher has a false-positive. But without these changes, it will break the wifi-tx on each match.

**Comment by hauke** [comment]
Could someone please update this pull request with the extra changes requested by @ecsv

**Comment by ecsv** [comment]
Looking at the PR - why was the 3.6 branch chosen here? Wouldn't it be better to use a consistent firmware (firmware-5.bin_10.4-*) for all wave2 cards (with exception of QCA99x0)? 3.5.3/firmware-5.bin_10.4-3.5.3-00053 seems to be the most used (QCA9888, QCA9984) firmware at the moment in package/firmware/ath10k-firmware/Makefile and in linux-firmware https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/log/ath10k/.

Wouldn't also be the worst idea to get the wave1 cards to the same firmware version. QCA9887 is currently using firmware-5.bin_10.2.4-1.0-00033 and QCA988X firmware-5.bin_10.2.4-1.0-00037. The latter was added to the repository with an "look, there is a new file in Kalle's repo" commit (349fe46103359682692e6b175d22f8c05ff75f74). 10.2.4-1.0-00037 is also the version shipped in linux-firmware

**Review by vort7u — COMMENTED** [review]
Changed from 3.6 to 3.5.3

File: package/firmware/ath10k-firmware/Makefile [hunk 1]
```diff
@@ -403,7 +403,7 @@ define Package/ath10k-firmware-qca4019/install
 		$(PKG_BUILD_DIR)/QCA4019/hw1.0/board-2.bin \
 		$(1)/lib/firmware/ath10k/QCA4019/hw1.0/
 	$(INSTALL_DATA) \
-		$(PKG_BUILD_DIR)/QCA4019/hw1.0/3.4/firmware-5.bin_10.4-3.4-00104 \
+		$(PKG_BUILD_DIR)/QCA4019/hw1.0/3.6/firmware-5.bin_10.4-3.6-00140 \
```
[the remark this answers is not in the document]
Comment (mastum) [reply package/firmware/ath10k-firmware/Makefile]: Sorry, I can not change the commit.
I tried the 3.5.3 for a few days and even with this I have no disconnections and warnings.

[the remark this answers is not in the document]
Comment (ecsv) [reply package/firmware/ath10k-firmware/Makefile]: you can. just use `git rebase -i d61c7466d772ea65c22d4d189d5e119f4adc3fb6~1` and squash the second commit in the first one

**Comment by Ansuel** [comment]
the atk10k dfs confirmation patch got merged...

think you should update all the firmware...

**Comment by blogic** [comment]
merged, Thanks !

**Comment by mastum** [comment]
Thanks @blogic, can you also apply on 18.06 branch? 😀

**Comment by blogic** [comment]
@Ansuel @mastum is that depending ath10k patch in 18.06 ?

**Comment by Ansuel** [comment]
Think it's only in master

**Comment by mastum** [comment]
> is that depending ath10k patch in 18.06 ?
@blogic no with firmware 3.5.3

**Comment by nuqleo** [comment]
With firmware 3.5.3 downloading for 5 GHZ clients ten times slower than with firmware 3.6.

**Comment by tapper82** [comment]
> 
> 
> With firmware 3.5.3 downloading for 5 GHZ clients ten times slower than with firmware 3.6.
Device? client? Tests ran?

**Comment by hauke** [comment]
Kalle send a pull request to update the ath10k firmware files in the linux-firmware repository recently:
http://lists.infradead.org/pipermail/ath10k/2018-October/012255.html
This contains the 3.6 FW and new board files for the QCA4019

We can update this in OpenWrt as soon as this was merged.

**Comment by chunkeey** [comment]
The problem with 3.6 firmware is that it needs a ath10k update too. I've merged this patch (with the name fixed)

**Pull request closed without merging**